portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -114,23 +129,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String text = request.getParameter(COMMENT_INPUT);
     long timestamp = System.currentTimeMillis();
     String name = (String) request.getParameter(COMMENT_NAME);
+    String nickname = HomeServlet.userNickname();
 
     if (name.isEmpty()) {
       name = ANONYMOUS_AUTHOR;
     }
-
     Entity commentEntity = new Entity(COMMENT_ENTITY);
     commentEntity.setProperty(COMMENT_TEXT, text);
     commentEntity.setProperty(COMMENT_TIMESTAMP, timestamp);
     commentEntity.setProperty(COMMENT_NAME, name);
     commentEntity.setProperty(COMMENT_LENGTH, text.length());
     commentEntity.setProperty(COMMENT_EMAIL, email);
+    commentEntity.setProperty(COMMENT_NICKNAME, nickname);
     
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
```suggestion
```

Already initialized.

Reply:
Thanks for catching that! I also removed the redundant code in NicknameServlet.

Done